### PR TITLE
Implements access to the porosity from Python.

### PR DESCRIPTION
**Pull request by hakonhagland**
Depends on #2830. 

Implements access (read/write) to the porosity from Python.

### CODE REVIEW

**Comment by hakonhagland** [comment]
jenkins build this please

File: opm/simulators/flow/python/simulators.hpp [hunk 1]
```diff
@@ -19,21 +16,32 @@
...
+
 #ifndef OPM_SIMULATORS_HEADER_INCLUDED
 #define OPM_SIMULATORS_HEADER_INCLUDED
 
 #include <opm/simulators/flow/Main.hpp>
 #include <opm/simulators/flow/FlowMainEbos.hpp>
+#include <opm/models/utils/propertysystem.hh>
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include <pybind11/embed.h>
```
Comment (joakim-hove) [inline opm/simulators/flow/python/simulators.hpp:28]: Should not need the `embed.h` header?

File: python/test/test_basic.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import unittest
+from opm2.simulators import BlackOilSimulator
+import os
+
+class TestBasic(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        os.chdir('test_data/SPE1CASE1')
+        # NOTE: Loading the below extension module involves loading a
```
Comment (joakim-hove) [inline python/test/test_basic.py:9]: Good description of an unfortunate situation

File: python/test/test_basic.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import unittest
+from opm2.simulators import BlackOilSimulator
+import os
+
+class TestBasic(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        os.chdir('test_data/SPE1CASE1')
```
Comment (joakim-hove) [inline python/test/test_basic.py:8]: There are two problems with the `os.chdir('test_data/SPE1CASE1')`:

1. When you hardcode the `test_data/SPE1CASE` that means the code *must* be run with cwd exactly at `python/test`. If you use `test_dir = os.path.join( os.path.dirname(__file__), "test_data/SPE1CASE1")` you are guaranteed to get the correct test directory irrespective of current cwd.

2. To change cwd changes the *state* of the process - that is ugly. With Python I recommend creating a small context:
```
from contextlib import contextmanager

@contextmanager
def pushd(path):
    cwd = os.getcwd()
    if not os.path.isdir(path):
        os.makedirs(path)
    os.chdir(path)

    yield

    os.chdir(cwd)
```
Then you can in your tests temporarily change directory with:
```
with pushd("/some/path/somewhere"):
      do_stuff_in_path()
      more_stuff()
# Now we are back in the original cwd
continue_in_cwd0()
```

File: python/simulators/CMakeLists.txt [hunk 4]
```diff
@@ -13,3 +19,21 @@ set(PYTHON_PACKAGE_PATH "site-packages")
...
+      DESTINATION ${PROJECT_BINARY_DIR}/python)
+file( COPY ${PROJECT_SOURCE_DIR}/python/test_data
+      DESTINATION ${PROJECT_BINARY_DIR}/python)
+
+find_program (BASH_EXECUTABLE bash)
+if (BASH_EXECUTABLE)
+  add_test(NAME python_tests
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/python
+    COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${PROJECT_BINARY_DIR}/python
+            ${BASH_EXECUTABLE} run-python-tests.sh )
```
Comment (joakim-hove) [inline python/simulators/CMakeLists.txt:38]: Could you not skip the entire `bash` script and use the cmake command:
```
COMMAND ${CMAKE_COMMAND} -E env PYTHONPATH=${PROJECT_BINARY_DIR/python:ENV{PYTHONPATH}
        ${PYTHON_EXECUTABLE} -m unittest
```

File: python/test/test_basic.py (hunk 3 above)
Comment (hakonhagland) [reply python/test/test_basic.py:8]: Good idea. But it seems the cwd needs to stay at `...test_data/SPE1CASE1` during the whole test, so if I use `with pushd..` in `setUpClass()` the cwd is reset when that finishes, and then `step_init()` throws `Exception received: Cannot find input case 'SPE1CASE1.DATA'.`

Comment (joakim-hove) [reply python/test/test_basic.py:8]: OK - the `pushd` should be in each of the test routines, whereas the `setUpClass` should just store the testpath.

**Comment by hakonhagland** [comment]
jenkins build this please

File: python/simulators/CMakeLists.txt (hunk 4 above)
Comment (hakonhagland) [reply python/simulators/CMakeLists.txt]: Yes! See the new commit.

**Comment by hakonhagland** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Then we merge - patient effort!

**Pull request merged**